### PR TITLE
Refactor generate_inverse_confusion_matrix

### LINKED ISSUE #1562: REM: Refactor generate_inverse_confusion_matrix
Per [PR feedback](https://github.com/unitaryfund/mitiq/pull/1449#discussion_r972863821) it was suggested to replace the use of the `NoisyReadoutSampler` and `cirq.measure_confusion_matrix` with a fixed analytical solution. Additionally, Revert changes from #1605.

As mentioned in the PR it is okay if this is a follow-up PR to #1449.

Here's a [reference](https://github.com/quantumlib/Cirq/pull/4854/files#diff-6037b8997ce87fcfcfe512167b87b8fb3b420061e5fd835f2fb2f3ce64554dd2R22)

### PR DESCRIPTION
**Pull request by amirebrahimi**
Resolves #1562

## Description

Replaced the use of `NoisyReadoutSampler` and `cirq.measure_confusion_matrix` with a straightforward approach of inverting a single-qubit confusion matrix and tensoring.

---

### License

- [X] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [ ] I added unit tests for new code.
- [X] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [X] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [ ] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [ ] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

**Comment by amirebrahimi** [comment]
Oddly enough, asv is failing here even though I removed the dependency, @Misty-W. I think it is because it compares against the main branch, no?

**Comment by Misty-W** [comment]
> Oddly enough, asv is failing here even though I removed the dependency, @Misty-W. I think it is because it compares against the main branch, no?

yes @amirebrahimi I saw the same thing when I tried to remove the dependency, and I think it's because the workflow compares against master.

**Review by andreamari — APPROVED** [review]
LGTM! Thanks @amirebrahimi for the quick PR! 

In addition to simplifying the the code, I am quite sure it will also fix the ASV failures that we are seeing everywhere. So I'll merge right now and check if it works. 

@natestemen, it seems that the ASV test in CI always pulls mitiq from the master branch even if the test is triggered by a PR commit (different branch).  Do you think this is normal behavior?

**Pull request merged**
